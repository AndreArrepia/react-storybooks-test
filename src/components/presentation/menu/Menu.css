.Menu{
    display: flex;
    width: 100%; 
    height: 80px;
    padding: 10px 15px;
    background-color: red;
    align-items: center;
    box-sizing: border-box;
}

.Menu h1{
    text-transform: uppercase;
    color: white;
}

.Menu ul{
    list-style-type: none;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
    flex: 1;
    height: 100%;
    margin-top: 0;
    margin-bottom: 0;
}

.Menu ul li{
    color: white;
    cursor: pointer;
    padding: 0 20px;
    border-radius: 6px;
    height: 100%;
    display: flex;
    align-items: center;
    text-transform: uppercase;
    font-family: 'Arial Balck', Courier, monospace;
    transition: background 0.5s ease-in-out, color 0.5s ease-in-out; 
}

.Menu ul li:hover{
    background: white;
    color: red;
}

@media only screen and (max-width: 1024px) {
    .Menu ul li{
        display: none;
    }
}



