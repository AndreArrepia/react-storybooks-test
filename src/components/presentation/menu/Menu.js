import React, { Component } from "react";
import PropTypes from "prop-types";

import "./Menu.css";

class Menu extends Component {
  constructor() {
    super();
    this.buildMenus = this.buildMenus.bind(this);
  }

  buildMenus(items) {
    return items.map((singleItem, index) => (
      <li key={index}> {singleItem.itemName} </li>
    ));
  }

  render() {
    return (
      <header className="Menu">
        <h1>{this.props.title}</h1>
        {this.props.menus &&
          this.props.menus.length > 0 && (
            <ul>{this.buildMenus(this.props.menus)}</ul>
          )}
      </header>
    );
  }
}

Menu.defaultProps = {
  title: "Hello World",
  menus: []
};

Menu.propTypes = {
  title: PropTypes.string,
  menus: PropTypes.array
};

export default Menu;
