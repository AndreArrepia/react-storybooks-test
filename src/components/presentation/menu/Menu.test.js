// Link.react.test.js
import React from "react";
import Menu from "./Menu";
import renderer from "react-test-renderer";

test("Menu Renders Without title and Menu Items", () => {
  const component = renderer.create(<Menu />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Menu Renders with title and without Menu Items", () => {
  const component = renderer.create(<Menu title="Custom title" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Menu Renders without title and with Menu Items", () => {
  const component = renderer.create(
    <Menu
      menus={[
        { itemName: "Item 1" },
        { itemName: "Item 2" },
        { itemName: "Item 3" }
      ]}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Menu Renders with title and Menu Items", () => {
  const component = renderer.create(
    <Menu
      title="Custom title"
      menus={[
        { itemName: "Item 1" },
        { itemName: "Item 2" },
        { itemName: "Item 3" }
      ]}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
