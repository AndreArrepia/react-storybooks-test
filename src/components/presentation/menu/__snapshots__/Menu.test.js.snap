// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Renders Without title and Menu Items 1`] = `
<header
  className="Menu"
>
  <h1>
    Hello World
  </h1>
</header>
`;

exports[`Menu Renders with title and Menu Items 1`] = `
<header
  className="Menu"
>
  <h1>
    Custom title
  </h1>
  <ul>
    <li>
       
      Item 1
       
    </li>
    <li>
       
      Item 2
       
    </li>
    <li>
       
      Item 3
       
    </li>
  </ul>
</header>
`;

exports[`Menu Renders with title and without Menu Items 1`] = `
<header
  className="Menu"
>
  <h1>
    Custom title
  </h1>
</header>
`;

exports[`Menu Renders without title and with Menu Items 1`] = `
<header
  className="Menu"
>
  <h1>
    Hello World
  </h1>
  <ul>
    <li>
       
      Item 1
       
    </li>
    <li>
       
      Item 2
       
    </li>
    <li>
       
      Item 3
       
    </li>
  </ul>
</header>
`;
